### PR TITLE
feat: postgresql support golang

### PR DESCRIPTION
**Pull request by CC11001100**
Hello, I used this project to do Postgresql parser in my recent work, but I found that there is currently support for Java and C#, no support for Golang, it took me some time to add support for golang and request push Merge into the original repository to help newcomers save time.

what I do:
1. Basic support of Postgresql's golang, some golang code file.
2. English and Chinese documents of how to use

### CODE REVIEW

File: sql/postgresql/Golang/README_zh.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Antlr v4 Postgresql Golang
```
Comment (KvanTTT) [inline sql/postgresql/Golang/README_zh.md:1]: We use only English across the entire repository. Please remove files in other languages. It is in low demand and almost nobody is able to understand and fix such files.

**Review by KvanTTT — CHANGES_REQUESTED** [review]

File: sql/postgresql/Golang/README_zh.md (hunk 1 above)
Comment (CC11001100) [reply sql/postgresql/Golang/README_zh.md]: delete README_zh.md:  https://github.com/antlr/grammars-v4/pull/2751/commits/dacbe9803f1c8d96b0c5fc904d02bc73193ad634

Comment (CC11001100) [reply sql/postgresql/Golang/README_zh.md]: Please have a look again

**Comment by kaby76** [comment]
[Just an FYI](https://github.com/antlr/grammars-v4/issues/2757#issuecomment-1219312856) about sub-parsing that is occurring for this grammar. Sub-parser code is not propagating errors to the main parse. After this code is merged, all target-specific code must be fixed to do this right.

Please note that the "Go" code should be in a directory called "Go" not "golang". The name of the directory is exactly the name in the target when specifying the Antlr4 Tool Jar run (e.g., `antlr4 -Dlanguage=Go -lib parser -o parser *.g4`). You can rename the directory now, or I will rename after you in a separate PR. Thanks.

**Comment by kaby76** [comment]
@CC11001100 Are you going to update your PR with the requested changes?

**Comment by CC11001100** [comment]
> @CC11001100 Are you going to update your PR with the requested changes?

Sorry for my poor English, I thought the merge was rejected. Thank you for your willingness to continue reviewing my code. I will keep modifying my PR until the merge is successful or rejected.

I renamed the `Golang ` directory to `Go`

**Review by KvanTTT — APPROVED** [review]

**Comment by CC11001100** [comment]
I see that the Java code for PG has changed. I was just translating the Java code to Go. It seems that I should modify the Go code until it behaves the same as the latest Java code? 

> [Just an FYI](https://github.com/antlr/grammars-v4/issues/2757#issuecomment-1219312856) about sub-parsing that is occurring for this grammar. Sub-parser code is not propagating errors to the main parse. After this code is merged, all target-specific code must be fixed to do this right.
> 
> Please note that the "Go" code should be in a directory called "Go" not "golang". The name of the directory is exactly the name in the target when specifying the Antlr4 Tool Jar run (e.g., `antlr4 -Dlanguage=Go -lib parser -o parser *.g4`). You can rename the directory now, or I will rename after you in a separate PR. Thanks.

**Comment by kaby76** [comment]
> I see that the Java code for PG has changed. I was just translating the Java code to Go. It seems that I should modify the Go code until it behaves the same as the latest Java code?

Yes, if you can model the changes that I made to the CSharp and Java targets, that would be great. Thank you so much for working on this. Also, your English is fine! --Ken

**Comment by kaby76** [comment]
Also, there is a huge fix to the Antlr Go runtime, to be released soon in version 4.11. The Go target performance will be as fast as any other target. So, the change here by you will be wonderful to see.

**Comment by teverett** [comment]
@kaby76  @KvanTTT  pull this?

**Comment by kaby76** [comment]
Yes, this is a good demo of the Go target.

**Comment by KvanTTT** [comment]
I've already approvedit.

**Comment by teverett** [comment]
@CC11001100 thanks!

**Pull request merged**